src/Development/IDE/Core/Compile.hs

```diff
@@ -582,7 +583,11 @@ parseFileContents customPreprocessor dflags comp_pkgs filename modTime contents
                        , pm_annotations = hpm_annotations
                       }
                    warnings = diagFromErrMsgs "parser" dflags warns
-               pure (warnings ++ preproc_warnings, pm)
+                   applyPluginAction p opts = parsedResultAction p opts ms
+               hspm <- liftIO $ runHsc env $ withPlugins dflags applyPluginAction 
+                 (HsParsedModule parsed' [] hpm_annotations)
+               let pm' = pm { pm_parsed_source = hpm_module hspm }
```

Comment:
This looks odd. Should we really be using hpm_module in other places?

Reply:
I had to convert back and forth between `HsParsedModule` and `ParsedModule`, because [`parsedResultAction`](https://hackage.haskell.org/package/ghc-8.10.1/docs/Plugins.html#v:parsedResultAction) deals with the former, whereas this function, `parseFileContents`, is dealing with the later.

This line is converting from the former (`hspm`) to the later (`pm'`), which is why we are pulling the parsed source from `hpm_source`. I'm not sure I understand why we would need to use it in other places?

Perhaps if I refactor it to be like this it would be more clear (transforming `parsed'` to `parsed''` where the plugin is applied)?

![image](https://user-images.githubusercontent.com/3998/93118127-cd18d600-f68d-11ea-84fb-a7da52f4970a.png)

Reply:
All of this code got moved to `D.I.G.Compat`.